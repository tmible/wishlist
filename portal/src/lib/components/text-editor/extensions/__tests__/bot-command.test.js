import { describe, expect, it, vi } from 'vitest';
import BotCommand from '../bot-command.js';

describe('text editor / extensions / bot command', () => {
  it('should be defined', () => {
    expect(BotCommand).toMatchSnapshot();
  });

  it('should parse HTML', () => {
    expect(BotCommand.config.parseHTML()).toMatchSnapshot();
  });

  it('should render HTML', () => {
    expect(BotCommand.config.renderHTML()).toMatchSnapshot();
  });

  describe('add commands', () => {
    it('should add set command', () => {
      const setMark = vi.fn();
      BotCommand.config.addCommands().setBotCommand()({ commands: { setMark } });
      expect(setMark).toHaveBeenCalledWith(BotCommand.name);
    });

    it('should add toggle command', () => {
      const toggleMark = vi.fn();
      BotCommand.config.addCommands().toggleBotCommand()({ commands: { toggleMark } });
      expect(toggleMark).toHaveBeenCalledWith(BotCommand.name);
    });

    it('should add unset command', () => {
      const unsetMark = vi.fn();
      BotCommand.config.addCommands().unsetBotCommand()({ commands: { unsetMark } });
      expect(unsetMark).toHaveBeenCalledWith(BotCommand.name);
    });
  });

  it('should add input rules', () => {
    expect(BotCommand.config.addInputRules()).toMatchSnapshot();
  });

  it('should add paste rules', () => {
    expect(BotCommand.config.addPasteRules()).toMatchSnapshot();
  });
});
