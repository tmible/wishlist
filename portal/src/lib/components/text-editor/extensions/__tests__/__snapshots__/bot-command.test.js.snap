// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`text editor / extensions / bot command > should add input rules 1`] = `
[
  InputRule {
    "find": /\\(\\?:\\^\\|\\\\s\\)\\(\\(\\\\/\\\\S\\+\\)\\\\s\\)\\$/,
    "handler": [Function],
  },
]
`;

exports[`text editor / extensions / bot command > should add paste rules 1`] = `
[
  PasteRule {
    "find": /\\(\\?:\\^\\|\\\\s\\)\\(\\(\\\\/\\\\S\\+\\)\\\\s\\)/g,
    "handler": [Function],
  },
]
`;

exports[`text editor / extensions / bot command > should be defined 1`] = `
Mark {
  "child": null,
  "config": {
    "addCommands": [Function],
    "addInputRules": [Function],
    "addPasteRules": [Function],
    "defaultOptions": {},
    "name": "bot_command",
    "parseHTML": [Function],
    "renderHTML": [Function],
  },
  "name": "bot_command",
  "options": {},
  "parent": null,
  "storage": {},
  "type": "mark",
}
`;

exports[`text editor / extensions / bot command > should parse HTML 1`] = `
[
  {
    "tag": "span[data-entity="bot_command"]",
  },
]
`;

exports[`text editor / extensions / bot command > should render HTML 1`] = `
[
  "span",
  {
    "class": "link cursor-text",
  },
  0,
]
`;
